powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/util/KeyDerivationUtils.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * PowerAuth Crypto Library
+ * Copyright 2019 Wultra s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.getlime.security.powerauth.crypto.lib.util;
+
+import io.getlime.security.powerauth.crypto.lib.config.PowerAuthConfiguration;
+import io.getlime.security.powerauth.crypto.lib.generator.KeyGenerator;
+import io.getlime.security.powerauth.crypto.lib.model.exception.GenericCryptoException;
+import io.getlime.security.powerauth.provider.exception.CryptoProviderException;
+
+import javax.crypto.SecretKey;
+import java.nio.ByteBuffer;
+import java.security.InvalidKeyException;
+
+/**
+ * The {@code KeyDerivationUtils} class provides additional key derivation functionality defined for PowerAuth protocol.
+ */
+public class KeyDerivationUtils {
+
+    private final KeyGenerator keyGenerator;
+
+    /**
+     * @param keyGenerator {@link KeyGenerator} instance to be used in the class.
+     */
+    public KeyDerivationUtils(KeyGenerator keyGenerator) {
+        this.keyGenerator = keyGenerator;
+    }
+
+    /**
+     * Derivation index used to derive KEY_TRANSPORT_IV from KEY_TRANSPORT
+     */
+    private static final long STATUS_BLOB_TRANSPORT_IV_INDEX = 3000;
```

Comment:
Shouldn't we document this in our developers portal? The value seems a bit magic here.

Reply:
See #291